src/transactions/TransactionFrame.cpp

```diff
@@ -349,7 +352,10 @@ TransactionFrame::processSignatures(SignatureChecker& signatureChecker,
         }
     }
 
-    removeUsedOneTimeSignerKeys(signatureChecker, ltxOuter);
+    if (ledgerVersion < 13)
+    {
+        removeOneTimeSignerFromAllSourceAccounts(ltxOuter);
```

Comment:
maybe we could add a `bool` property to `signatureChecker` (set where we used to add to the map of used one time signatures) to know if one time signers were used at all to decide if we want to call that function?
The new code is always going to loop over all accounts / all signers (so loading up to 101 accounts and perform `101*20=2020` extra checks for no reason)

Reply:
I don't think this makes sense. In the event that we aren't checking signatures (eg. `cv < ValidationType::kInvalidPostAuth`) then we have to loop over all accounts and signers, since they haven't been checked by `SignatureChecker`. In the event that we are checking signatures (eg. `cv >= ValidationType::kInvalidPostAuth`) then we have already loaded the accounts in the `SignatureChecker` so they are in memory and looping over the accounts and signers requires no I/O.

In the worst case, we have to do the full loop including I/O. In all other cases, we have already done the I/O so it shouldn't be expensive. If at a future date we assess that this is an unacceptable amount of overhead, let's optimize and make the code more complicated at that point.

Reply:
I guess this was more a comment on the fact that in the previous version we had 0 overhead when not using one time signers to an implementation that bears the overhead all the time (all that for a feature that is virtually never used).

I am quite not sure what the overhead of this is: maintaining `LedgerTxn` is not free, we had examples in the past where the overhead was managing maps.

I guess what we can do is perform some performance test comparison of this new implementation to compare it against 12.5.0, replaying protocol 12 ledger ranges with version 13.0.0 both with protocol 12 and 13.